modules/code-editor/src/backend/definitions.ts

```diff
@@ -22,7 +23,7 @@ export interface FileDefinition {
   /** Validation if the selected file can be deleted */
   canDelete?: (file: EditableFile) => boolean
   /** An additional validation that must be done for that type of file. Return a string indicating the error message */
-  validate?: (file: EditableFile) => Promise<string | undefined>
+  validate?: (file: EditableFile, isWriting?: boolean) => Promise<string | undefined>
```

Comment:
Always feels a bit weird to pass bools as function params, could you split this in 2 functions ?  or does it even make sens to pass the bool ? can you check the value before calling the hook ? and maybe prevent the call ?

Reply:
Can't check the value before, it's the backend verification and must be done at that location. The reason I've added the "validate" method is so we only have to configure every file type at one location (the file definitions) without having to add custom logic at different location for different type of files. 

This allows the usage of a single middleware, which validates the payload sent by the user before calling any method on the backend to save the content.